In [1]:
import pandas as pd 
purchase_data_df = "data/purchase_data.csv" #Purchasing file location
purchase_data_df = pd.read_csv(purchase_data_df) # Reads Purchasing File and store into Pandas data frame
purchase_data_df #displays purchase data dataframe

Purchase ID             SN  Age  Gender  Item ID  \
0              0        Lisim78   20    Male      108   
1              1    Lisovynya38   40    Male      143   
2              2     Ithergue48   24    Male       92   
3              3  Chamassasya86   24    Male      100   
4              4      Iskosia90   23    Male      131   
..           ...            ...  ...     ...      ...   
775          775     Aethedru70   21  Female       60   
776          776         Iral74   21    Male      164   
777          777     Yathecal72   20    Male       67   
778          778        Sisur91    7    Male       92   
779          779      Ennrian78   24    Male       50   

                                     Item Name  Price  
0    Extraction, Quickblade Of Trembling Hands   3.53  
1                            Frenzied Scimitar   1.56  
2                                 Final Critic   4.88  
3                                  Blindscythe   3.27  
4                                         Fury   1.44  
..                                         ...    ...  
775                                       Wolf   3.54  
776                           Exiled Doomblade   1.63  
777      Celeste, Incarnation of the Corrupted   3.46  
778                               Final Critic   4.19  
779                                       Dawn   4.60  

[780 rows x 7 columns]

## Player Count

* Displays the total number of players


In [2]:
#creates dataframe for TOTAL NUMBER OF INDIVIDUAL PLAYERS
tot_players_df=pd.DataFrame({
    "Total Players":[purchase_data_df['SN'].nunique()]
})
tot_players_df #display total players dataframe

Total Players
0            576

## Purchasing Analysis (Total)

In [3]:
#create dataframe for purchasing analysis
#total count = number of unique items
#average price per player = mean of price
#number of purchases = count of purchase id
#total revenue = sum of price
purchasing_analysis_total_df=pd.DataFrame({
    "Total Count":[purchase_data_df['Item Name'].nunique()],
    "Average Price Per Player":[purchase_data_df['Price'].mean()],
    "Number of Purchases":[purchase_data_df['Purchase ID'].count()],
    "Total Revenue":[purchase_data_df['Price'].sum()]
})
purchasing_analysis_total_df['Average Price Per Player']=purchasing_analysis_total_df['Average Price Per Player'].map("${:.2f}".format) #formats column as currency
purchasing_analysis_total_df['Total Revenue']=purchasing_analysis_total_df['Total Revenue'].map("${:.2f}".format) #formats column as currency

purchasing_analysis_total_df #displays dataframe

Total Count Average Price Per Player  Number of Purchases Total Revenue
0          179                    $3.05                  780      $2379.77

## Gender Demographics

In [4]:
gender_group_gb=purchase_data_df.groupby(["Gender"]) #creates groupby object for GENDER
gender_count_ser=gender_group_gb['SN'].nunique() #counts number of unique players based on gender
gender_perc_players_ser=gender_count_ser/tot_players_df['Total Players'][0] #calculates percent of players for each gender

#creates dataframe for gender count and percentages
gender_dem_df=pd.DataFrame({"Total Count":gender_count_ser,
                             "Percentage of Players":gender_perc_players_ser    
})

gender_dem_df['Percentage of Players']=gender_dem_df['Percentage of Players']*100 #formats column as percentage
gender_dem_df['Percentage of Players']=gender_dem_df['Percentage of Players'].map("{:.2f}%".format) #formats column as percentage
gender_dem_df.rename_axis(None,inplace=True) #removes index title for Gender
gender_dem_df #displays dataframe

Total Count Percentage of Players
Female                          81                14.06%
Male                           484                84.03%
Other / Non-Disclosed           11                 1.91%


## Purchasing Analysis (Gender)

In [5]:
num_of_players_gender=gender_group_gb['SN'].nunique() #counts number of unique players based on gender
purchase_count_gender_ser=gender_group_gb['Purchase ID'].count() #counts number of purchases for each gender
tot_purchase_val_gender_ser=gender_group_gb['Price'].sum() #sums all purchases for each gender
avg_purchase_price_gender_ser=tot_purchase_val_gender_ser/purchase_count_gender_ser #averages purchase price for each person/gender based on number of purchases
avg_purchase_price_per_person_gender_ser=tot_purchase_val_gender_ser/num_of_players_gender #averages purchase price for each person/gender based on number of individual players

#creates dataframe
purchasing_analysis_gender_df=pd.DataFrame({"Purchase Count":purchase_count_gender_ser,
                            "Average Purchase Price":avg_purchase_price_gender_ser,
                            "Total Purchase Value":tot_purchase_val_gender_ser,
                            "Avg Total Purchase Per Person":avg_purchase_price_per_person_gender_ser
})

#formats dataframe
purchasing_analysis_gender_df['Average Purchase Price']=purchasing_analysis_gender_df['Average Purchase Price'].map("${:.2f}".format)
purchasing_analysis_gender_df['Total Purchase Value']=purchasing_analysis_gender_df['Total Purchase Value'].map("${:.2f}".format)
purchasing_analysis_gender_df['Avg Total Purchase Per Person']=purchasing_analysis_gender_df['Avg Total Purchase Per Person'].map("${:.2f}".format)

purchasing_analysis_gender_df.rename_axis(None,inplace=True) #removes index title for Gender
purchasing_analysis_gender_df #displays dataframe

Purchase Count Average Purchase Price  \
Female                            113                  $3.20   
Male                              652                  $3.02   
Other / Non-Disclosed              15                  $3.35   

                      Total Purchase Value Avg Total Purchase Per Person  
Female                             $361.94                         $4.47  
Male                              $1967.64                         $4.07  
Other / Non-Disclosed               $50.19                         $4.56

## Age Demographics

In [6]:
temp_group=purchase_data_df.groupby('SN') #creates groupby object based on screen names
temp_group=temp_group['Age'].mean() #attains age of each player by taking average of their age

bins=[0,9,14,19,24,29,34,39,150] #creates bin ranges
bin_names=["<10","10-14","15-19","20-24","25-29","30-34","35-39","40+"] #creates bin labels
tot_count_age=pd.cut(temp_group, bins, labels=bin_names, include_lowest=True) #bins the ages for each player

tot_count=tot_count_age.value_counts() #counts number people in each bin
perc_of_players=tot_count/tot_players_df['Total Players'][0] #percentages of players based on age

#creates dataframe
age_summary_df=pd.DataFrame({"Total Count":tot_count,
                             "Percent of Players":perc_of_players
 })

#formats cells
age_summary_df['Percent of Players']=age_summary_df["Percent of Players"]*100
age_summary_df['Percent of Players']=age_summary_df["Percent of Players"].map("{:.2f}%".format)
age_summary_df.sort_index()

Total Count Percent of Players
<10             17              2.95%
10-14           22              3.82%
15-19          107             18.58%
20-24          258             44.79%
25-29           77             13.37%
30-34           52              9.03%
35-39           31              5.38%
40+             12              2.08%

## Purchasing Analysis (Age)

In [33]:
bins=[0,9,14,19,24,29,34,39,150] #creates bin ranges
bin_labels=["<10","10-14","15-19","20-24","25-29","30-34","35-39","40+"] #creates bin labels
categorized_age_ser=pd.cut(purchase_data_df['Age'], bins, labels=bin_labels, include_lowest=True) #bins by age

purchase_data_df['Categorized Age']=categorized_age_ser #creates column for binned ages

ages_group_gb=purchase_data_df.groupby('Categorized Age') #groups by categorized ages
purchase_count_age_ser=ages_group_gb['Purchase ID'].count() #counts of purchased based on binned ages
purchase_count_age_ser

avg_purchase_price_age_ser=ages_group_gb['Price'].mean() #averages the prices based on binned ages
avg_purchase_price_age_ser

total_purchase_val_age_ser=ages_group_gb['Price'].sum() #sums of prices based on binned ages
total_purchase_val_age_ser

num_of_players_age=ages_group_gb['SN'].nunique() #diplays number of individual players for each age group
num_of_players_age

avg_total_purchase_per_person_age_ser=ages_group_gb['Price'].sum()/num_of_players_age #averages the purchase price based on individual players
avg_total_purchase_per_person_age_ser

#creates dataframe
purchase_analysis_age_df=pd.DataFrame({"Purchase Count":purchase_count_age_ser,
                                       "Average Purchase Price":avg_purchase_price_age_ser,
                                       "Total Purchase Value":total_purchase_val_age_ser,
                                       "Avg Total Purchase per Person":avg_total_purchase_per_person_age_ser
 })

#formats dataframe
purchase_analysis_age_df['Average Purchase Price']=purchase_analysis_age_df["Average Purchase Price"].map("${:.2f}".format)
purchase_analysis_age_df['Total Purchase Value']=purchase_analysis_age_df["Total Purchase Value"].map("${:.2f}".format)
purchase_analysis_age_df['Avg Total Purchase per Person']=purchase_analysis_age_df["Avg Total Purchase per Person"].map("${:.2f}".format)

purchase_analysis_age_df.rename_axis(None,inplace=True) #removes index title for categorized ages
purchase_analysis_age_df

Purchase Count Average Purchase Price Total Purchase Value  \
<10                23                  $3.35               $77.13   
10-14              28                  $2.96               $82.78   
15-19             136                  $3.04              $412.89   
20-24             365                  $3.05             $1114.06   
25-29             101                  $2.90              $293.00   
30-34              73                  $2.93              $214.00   
35-39              41                  $3.60              $147.67   
40+                13                  $2.94               $38.24   

      Avg Total Purchase per Person  
<10                           $4.54  
10-14                         $3.76  
15-19                         $3.86  
20-24                         $4.32  
25-29                         $3.81  
30-34                         $4.12  
35-39                         $4.76  
40+                           $3.19

## Top Spenders

In [32]:
sn_group_gb=purchase_data_df.groupby('SN') #creates groupby object based on screen name

purchase_count_sn_ser=sn_group_gb['Purchase ID'].count() #counts transactions for each player
purchase_count_sn_ser

avg_purchase_price_sn_ser=sn_group_gb['Price'].mean() #averages price paid for each player
avg_purchase_price_sn_ser

total_purchase_value_sn_ser=sn_group_gb['Price'].sum() #totals of all transactions per player
total_purchase_value_sn_ser

#creates dataframe
top_spenders_df=pd.DataFrame({"Purchase Count":purchase_count_sn_ser,
                                       "Average Purchase Price":avg_purchase_price_sn_ser,
                                       "Total Purchase Value":total_purchase_value_sn_ser
 })

#formats dataframe
top_spenders_df['Average Purchase Price']=top_spenders_df["Average Purchase Price"].map("${:.2f}".format)
top_spenders_df['Total Purchase Value']=top_spenders_df["Total Purchase Value"].map("${:.2f}".format)
top_spenders_df=top_spenders_df.sort_values(by=['Purchase Count'], ascending=False)
top_spenders_df.rename_axis(None,inplace=True)
top_spenders_df.head()


Purchase Count Average Purchase Price Total Purchase Value
Lisosia93                 5                  $3.79               $18.96
Iral74                    4                  $3.40               $13.62
Idastidru52               4                  $3.86               $15.45
Asur53                    3                  $2.48                $7.44
Inguron55                 3                  $3.70               $11.11

## Most Popular Items

In [22]:
most_popular_item_raw_df=purchase_data_df[["Purchase ID",'Item ID',"Item Name","Price"]] #creates new dataframe based on original dataframe

most_popular_item_group_gb=most_popular_item_raw_df.groupby(["Item ID","Item Name"]) #groups object based on item id and the item name

purchase_count_popular_ser=most_popular_item_group_gb["Purchase ID"].count() #number count of times an item was purchased
purchase_count_popular_ser

item_price_popular_ser=most_popular_item_group_gb["Price"].mean() #average paid for each item 
item_price_popular_ser

total_purchase_value_popular_ser=most_popular_item_group_gb["Price"].sum() #total paid for each item
total_purchase_value_popular_ser

#creates dataframe
most_popular_df=pd.DataFrame({"Purchase Count":purchase_count_popular_ser,
                                       "Item Price":item_price_popular_ser,
                                       "Total Purchase Value":total_purchase_value_popular_ser
 })

#formats dataframe
most_popular_df['Item Price']=most_popular_df['Item Price'].map("${:.2f}".format)
most_popular_df['Total Purchase Value']=most_popular_df["Total Purchase Value"].map("${:.2f}".format)
most_popular_df=most_popular_df.sort_values(by=['Purchase Count'], ascending=False)
most_popular_df.head()

Purchase Count  \
Item ID Item Name                                                      
92      Final Critic                                              13   
178     Oathbreaker, Last Hope of the Breaking Storm              12   
145     Fiery Glass Crusader                                       9   
132     Persuasion                                                 9   
108     Extraction, Quickblade Of Trembling Hands                  9   

                                                     Item Price  \
Item ID Item Name                                                 
92      Final Critic                                      $4.61   
178     Oathbreaker, Last Hope of the Breaking Storm      $4.23   
145     Fiery Glass Crusader                              $4.58   
132     Persuasion                                        $3.22   
108     Extraction, Quickblade Of Trembling Hands         $3.53   

                                                     Total Purchase Value  
Item ID Item Name                                                          
92      Final Critic                                               $59.99  
178     Oathbreaker, Last Hope of the Breaking Storm               $50.76  
145     Fiery Glass Crusader                                       $41.22  
132     Persuasion                                                 $28.99  
108     Extraction, Quickblade Of Trembling Hands                  $31.77

## Most Profitable Items

In [10]:
#creates dataframe using same materials as above
most_profitable_df=pd.DataFrame({"Purchase Count":purchase_count_popular_ser,
                                       "Item Price":item_price_popular_ser,
                                       "Total Purchase Value":total_purchase_value_popular_ser
 }).sort_values(by=['Total Purchase Value'], ascending=False) #sorts dataframe

#formats dataframe
most_profitable_df['Item Price']=most_profitable_df['Item Price'].map("${:.2f}".format)
most_profitable_df['Total Purchase Value']=most_profitable_df["Total Purchase Value"].map("${:.2f}".format)

most_profitable_df

Purchase Count  \
Item ID Item Name                                                      
92      Final Critic                                              13   
178     Oathbreaker, Last Hope of the Breaking Storm              12   
82      Nirvana                                                    9   
145     Fiery Glass Crusader                                       9   
103     Singed Scalpel                                             8   
...                                                              ...   
28      Flux, Destroyer of Due Diligence                           2   
125     Whistling Mithril Warblade                                 2   
126     Exiled Mithril Longsword                                   1   
104     Gladiator's Glaive                                         1   
42      The Decapitator                                            1   

                                                     Item Price  \
Item ID Item Name                                                 
92      Final Critic                                      $4.61   
178     Oathbreaker, Last Hope of the Breaking Storm      $4.23   
82      Nirvana                                           $4.90   
145     Fiery Glass Crusader                              $4.58   
103     Singed Scalpel                                    $4.35   
...                                                         ...   
28      Flux, Destroyer of Due Diligence                  $1.06   
125     Whistling Mithril Warblade                        $1.00   
126     Exiled Mithril Longsword                          $2.00   
104     Gladiator's Glaive                                $1.93   
42      The Decapitator                                   $1.75   

                                                     Total Purchase Value  
Item ID Item Name                                                          
92      Final Critic                                               $59.99  
178     Oathbreaker, Last Hope of the Breaking Storm               $50.76  
82      Nirvana                                                    $44.10  
145     Fiery Glass Crusader                                       $41.22  
103     Singed Scalpel                                             $34.80  
...                                                                   ...  
28      Flux, Destroyer of Due Diligence                            $2.12  
125     Whistling Mithril Warblade                                  $2.00  
126     Exiled Mithril Longsword                                    $2.00  
104     Gladiator's Glaive                                          $1.93  
42      The Decapitator                                             $1.75  

[179 rows x 3 columns]

## Written Analysis 
* Approximately 80% of the players identify themselves as male, accounting for 484 players out of the 576 total population of players. As such, the males account for approximately 83% of total purchases. However, females have a slightly greater average purchase price and average purchase price per person. This could indicate that more men play the game but spend less on the game.

* By examining the purchase counts there seems to be a positive correlation between the item price and the number of times it is purchased. 

* By examining the top spenders we can see that most people are one time customers (i.e., they have only bought one item). 




